src/main/java/org/fenixedu/academic/domain/Professorship.java

```diff
@@ -88,6 +94,7 @@ public void delete() {
         ProfessorshipManagementLog.createLog(getExecutionCourse(), Bundle.MESSAGING, "log.executionCourse.professorship.removed",
                 getPerson().getPresentationName(), getExecutionCourse().getNome(), getExecutionCourse()
                         .getDegreePresentationString());
+        ExecutionCourse executionCourse = getExecutionCourse();
```

Comment:
The idea here was to trigger a signal on deletion, but currently that is not possible (the object will not exist on signal handler invocation). Pull request updated to remove these lines.